lazy_dataset/core.py

```diff
@@ -2313,6 +2397,95 @@ def __iter__(self):
             print(f'Dropped {dropped_count} examples')
 
 
+class CacheDataset(Dataset):
+    def __init__(self, dataset: Dataset, keep_mem_free=None,
+                 immutable_warranty: str = 'pickle') -> None:
+        super().__init__()
+        assert dataset.indexable, (
+            'Cache Dataset only works if dataset is indexable!'
+        )
+        self.dataset = dataset
+        self.cache = {}
+
+        self._keep_mem_free = self._get_memory_size(keep_mem_free)
+        self.immutable_warranty = immutable_warranty
+
+        if immutable_warranty == 'pickle':
+            self._serialize = pickle.dumps
+            self._deserialize = pickle.loads
+        elif immutable_warranty == 'copy':
+            self._serialize = lambda x: x
+            self._deserialize = deepcopy
+        else:
+            raise ValueError(immutable_warranty)
+
+    @property
+    def indexable(self) -> bool:
+        return True
```

Comment:
Can you return `dataset.indexable`? In the long term we may weaken that indexable is necessary (Orderd is enough, but the flag is not jet implemented)

Reply:
Yes

Reply:
Done